integration-tests/slack/notify-circleci-test-results.js

```diff
@@ -0,0 +1,77 @@
+const _ = require('lodash/fp')
+const { getFailedTestNamesFromArtifacts } = require('./circleci-utils')
+const { getMessageBlockTemplate } = require('./message-templates')
+const { postMessage } = require('./post-message')
+const testsInfo = require('./slack-notify-channels.json')
+
+
+/**
+ * Get array of Slack channel IDs for succeeded job notification
+ * @returns {Array[string]}
+ */
+const getAllSlackChannelsForPassedJob = () => {
+  return _.map(_.get('id'), testsInfo.pass)
+}
+
+/**
+ * Get all Slack channel IDs and test names for failed job notification
+ * @returns {Map<string, Array[string]>}
+ */
+const getAllSlackChannelsForFailedJob = () => {
+  const allChannelsAndTests = new Map()
+  _.forEach(item => {
+    const channelId = _.get('id', item)
+    const testNames = _.map(test => test.name, item.tests)
+    allChannelsAndTests.set(channelId, testNames)
+  }, testsInfo.fail)
+  return allChannelsAndTests
```

Comment:
Nit: this could probably also be written something like:
```suggestion
  return _.fromPairs(_.map(item => [item.id, _.map('name', item.tests)], testsInfo.fail))
```

Reply:
Unchanged after tested this because it returns a Dictionary instead of a Map object.